lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -102,6 +102,33 @@ def cleanup(runner_context)
           delete_secret(secret) if secret
         end
 
+        def wait(timeout: nil, events: %i[create update delete])
+          watcher = kubeclient.watch_pods(:namespace => namespace)
+
+          timeout_thread = Thread.new { sleep(timeout); watcher.finish } if timeout.to_i > 0
+
+          watcher.each do |notice|
+            event = kube_notice_type_to_event(notice.type)
+            next unless events.include?(event)
+
+            pod = notice.object
+            container_ref   = pod.metadata.name
+            container_state = pod.to_h.deep_stringify_keys["status"]
+
+            runner_context = {"container_ref" => container_ref, "container_state" => container_state}
+
+            if block_given?
+              yield runner_context
+            else
+              timeout_thread&.kill # If we break out before the timeout, kill the timeout thread
+              return runner_context
```

Comment:
I noticed that the docker one builds up an array of notices and returns all of them, whereas this one only returns a single context.  That means the return interface is inconsistent (Array of Hash vs just a single Hash).  Is that intentional because of how kube works? Or perhaps this should return `[runner_context]`?

Reply:
Changed this to return `[]`, did the kubernetes one first then found docker events could return multiple.  They both yield individual items at a time but kubernetes can't return multiple.